packages/bsky/src/api/app/bsky/graph/getList.ts

```diff
@@ -68,9 +70,41 @@ const hydration = async (
       params.hydrateCtx,
     ),
   ])
+
+  const list = listState.lists?.get(listUri)
+  const ownerDid = new AtUri(listUri).hostname
+  if (
+    list?.record.purpose === 'app.bsky.graph.defs#referencelist' &&
+    // We show all users regardless of blocks if the viewer is the owner of the list, so no need to hydrate them
+    params.hydrateCtx.viewer !== ownerDid
+  ) {
+    const pairs: Map<string, RelationshipPair> = new Map()
+    for (const { did } of listitems) {
+      pairs.set(did, [did, ownerDid])
+    }
+    const bidirectionalBlocks =
+      await ctx.hydrator.hydrateBidirectionalBlocks(pairs)
+    return mergeStates(listState, { ...profileState, bidirectionalBlocks })
+  }
+
   return mergeStates(listState, profileState)
 }
 
+const noBlocks = (input: RulesFnInput<Context, Params, SkeletonState>) => {
+  const { skeleton, hydration } = input
+  if (!hydration.bidirectionalBlocks) {
+    return skeleton
+  }
+  skeleton.listitems = skeleton.listitems.filter(({ did }) => {
+    const blocks = hydration.bidirectionalBlocks?.get(did)?.values() ?? []
+    for (const block of blocks || []) {
```

Comment:
Already got the default rockin above.
```suggestion
    for (const block of blocks) {
```